clients/src/utils/encryption/index.ts

```diff
@@ -2,19 +2,45 @@ import * as crypto from 'crypto';
 import * as ed from 'noble-ed25519';
 import * as bs58 from 'bs58';
 
-export const createChallenge = (): string => {
-	const challenge = crypto.randomBytes(30).toString('hex');
-	return crypto.createHash('sha256').update(challenge).digest().toString();
+export const createNonce = (): string => {
+	return crypto.randomBytes(20).toString('hex');
 };
 
-export const getHexEncodedKey = (base58Key: string) => {
+export const getHexEncodedKey = (base58Key: string): string => {
 	return bs58.decode(base58Key).toString('hex');
 };
 
-export const signChallenge = async (privateKey: string, challenge: string) => {
-	return await ed.sign(challenge, privateKey);
+const hashNonce = (nonce: string) => crypto.createHash('sha256').update(nonce).digest().toString();
+
+export const signNonce = async (privateKey: string, nonce: string): Promise<string> => {
```

Comment:
Similar to the `api` folder, I would check length of the params

Reply:
ha ok I had this but removed it because I was unclear if I should to do so :P